### ISSUE TITLE
[BUG] Cannot provide controls u for model.simulate

### ISSUE THREAD

**Issue by juliuskittler** [dynamicslab/pysindy#92]
After fitting a continuous-time SINDy model with controls `u_train`, I am trying to simulate with `u_test`. This raises an error message. `TypeError: 'numpy.ndarray' object is not callable`

It seems like something is wrong with the nested if-statements in the beginning of the simulate function in:
https://github.com/dynamicslab/pysindy/blob/master/pysindy/pysindy.py Somehow, I am ending up at a condition, where the code is trying to call my `u_test` as a function although it is an array.

### Reproducing code example:

```python
import numpy as np
from scipy.integrate import odeint
import pysindy as ps

def lorenz(z, t):
    return [
        10 * (z[1] - z[0]),
        z[0] * (28 - z[2]) - z[1],
        z[0] * z[1] - (8 / 3) * z[2]
    ]

# Generate training data
dt = .002
t_train = np.arange(0, 10, dt)
x0_train = [-8, 8, 27]
x_train = odeint(lorenz, x0_train, t_train)
u_train = np.ones(x_train.shape[0]) # RELEVANT ROW

# Evolve the Lorenz equations in time using a different initial condition
t_test = np.arange(0, 15, dt)
x0_test = np.array([8, 7, 15])
x_test = odeint(lorenz, x0_test, t_test)
u_test = np.ones(x_test.shape[0]) # RELEVANT ROW

# Initialize model
model = ps.SINDy()

# Fit model
model.fit(x = x_train, u = u_train, t = t_train)

# Simulate
model.simulate(x0 = x0_test, u = u_test, t = t_test)

```
### Error message:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-99-29456ec7cb28> in <module>
     31 
     32 # Simulate
---> 33 model.simulate(x0 = x0_test, u = u_test, t = t_test)

~/.cache/pypoetry/virtualenvs/powerup-process-control-VY7sLJ9r-py3.8/lib/python3.8/site-packages/pysindy/pysindy.py in simulate(self, x0, t, u, integrator, stop_condition, **integrator_kws)
    700 
    701             else:
--> 702                 if ndim(u(1)) == 1:
    703 
    704                     def rhs(x, t):

TypeError: 'numpy.ndarray' object is not callable
```

### PySINDy/Python version information:

1.2.0

**Comment by briandesilva** [comment]
Since `SINDy.simulate` uses an adaptive integrator by default, it requires the control input to be a function rather than an array. See [this example](https://pysindy.readthedocs.io/en/latest/examples/1_feature_overview.html#simulate-forward-in-time-unknown-control-input-function) for a quick workaround.

It's confusing that the control input is a vector for the `fit` method, but not for the `simulate` method. I'll update `simulate` to handle both types of inputs.

**Comment by juliuskittler** [comment]
@briandesilva, thanks a lot for getting back. This workaround is helpful.

Yes, I was looking at the documentation of the `simulate` function and it said that it should take an array as well. Hence, my confusion. `u: function from R^1 to R^{n_control_features} or list/array` Thanks for taking a look at it!

**Comment by briandesilva** [comment]
> Yes, I was looking at the documentation of the simulate function and it said that it should take an array as well.

That's a mistake on our part. Sorry for the confusion!

**Comment by briandesilva** [comment]
Fixed in fa42b1bc0098a8c5b2a677a365afc35f0f9b8f25.
